### ISSUE TITLE
Vulnerable dependencies being treated only as "outdated components"

### ISSUE THREAD

**Issue by idan89** [DependencyTrack/dependency-track#686]
Hello, I have tested a bom.xml sample file with known vulnerabilities but none of them were identified. 
I've tested it with and without OSSIndex analyzer active.

For example, rake version 11.3.0 has known vulnerabilities [here](https://nvd.nist.gov/vuln/detail/CVE-2020-8130).

![image](https://user-images.githubusercontent.com/6001184/82678633-4fcddf00-9c20-11ea-9170-aaa015ecc62c.png)

But is only being treated as a **Risk - Outdated Component**

Am I doing something wrong here?
Thank you.

[bom.xml.zip](https://github.com/DependencyTrack/dependency-track/files/4668656/bom.xml.zip)

**Comment by stevespringett** [comment]
According to OSS Index, it's vulnerable.
https://ossindex.sonatype.org/component/pkg:gem/rake@11.3.0

So, that tells me that the BOM was likely imported prior to OSS Index analyzer being enabled, or there was an issue with communication to OSS Index.

Ensure that OSS Index is enabled, that you're not getting any errors in the logs when you submit the BOM.

**Comment by idan89** [comment]
Thank you for the quick response @stevespringett.
OSSIndex is active.  I tried even by creating a new project.
Here is the log when uploading the bom.xml file:

```
16:02:31.870 INFO [BomUploadProcessingTask] Processing CycloneDX BOM uploaded to project: 60533b5a-19f1-4233-8b8c-5e6bd719311f
16:02:32.447 INFO [BomUploadProcessingTask] Processed 93 components uploaded to project 60533b5a-19f1-4233-8b8c-5e6bd719311f
16:02:32.457 INFO [InternalAnalysisTask] Starting internal analysis task
16:02:32.458 INFO [MetricsUpdateTask] Executing metrics update for project: 60533b5a-19f1-4233-8b8c-5e6bd719311f
16:02:32.458 INFO [OssIndexAnalysisTask] Starting Sonatype OSS Index analysis task
16:02:32.461 INFO [InternalAnalysisTask] Internal analysis complete
16:02:32.461 INFO [MetricsUpdateTask] Executing metrics update for project: 60533b5a-19f1-4233-8b8c-5e6bd719311f
16:02:32.792 INFO [MetricsUpdateTask] Completed metrics update for project: 60533b5a-19f1-4233-8b8c-5e6bd719311f
16:02:32.820 INFO [MetricsUpdateTask] Completed metrics update for project: 60533b5a-19f1-4233-8b8c-5e6bd719311f
16:02:32.994 INFO [OssIndexAnalysisTask] Analyzing 93 component(s)
16:02:37.994 INFO [OssIndexAnalysisTask] Sonatype OSS Index analysis complete
16:02:37.995 INFO [MetricsUpdateTask] Executing metrics update for project: 60533b5a-19f1-4233-8b8c-5e6bd719311f
16:02:38.256 INFO [MetricsUpdateTask] Completed metrics update for project: 60533b5a-19f1-4233-8b8c-5e6bd719311f
```

Despite the fact that I don't see any errors or warnings, there is no vulnerability being identified. 
Thank you in advance.

**Comment by stevespringett** [comment]
I'm not sure why. I imported the BOM provided into my test instance and it was able to identify many vulnerabilities from the components, including rake.

<img width="1579" alt="Screen Shot 2020-05-22 at 11 13 54 AM" src="https://user-images.githubusercontent.com/3878933/82687899-90782900-9c1d-11ea-928e-aefc36d3e604.png">

So the BOM is certainly good. Must be a configuration or environmental issue. Try turning on debug logging. That may provide additional insight.

**Comment by idan89** [comment]
I've found the problem. My OSSIndex account was not "activated" (I did not confirm the email activation process), even when I logged in and retrieved the OSSIndex API Token information for setting up the integration with DT.
After that the vulnerabilities were identified by DT.
![image](https://user-images.githubusercontent.com/6001184/82688079-a510ed00-9c2e-11ea-8884-e90e1ea97284.png)


It would be useful if an error or warning is shown within the logs.
Thank you @stevespringett, I will close this issue.
